### DISCUSSION TITLE
HowTo: OMX HWA in docker on RPI4 Ubuntu Server 64 bit Host

### DISCUSSION THREAD

**Discussion by whiteowl3** [jellyfin/jellyfin#6215] [Show and tell]
# EDITED: July 1, 2021
Cursory use of the lastest version suggests that this info may be obsolete.  I haven't looked at the change history to confirm,
this edit is just to say that you should try the settings from the lsio guide before implementing any of this.  I will update after I am able to test.

# Jellyfin OMX HWA in docker on RPI4 Ubuntu Server 64 bit Host:
 
#### references:
https://docs.linuxserver.io/images/docker-jellyfin
https://github.com/jellyfin/jellyfin/issues/4892
https://github.com/linuxserver/docker-jellyfin/issues/14
 
### on your host, you will need to (with root):
 
dpkg --add-architecture armhf # Allow apt to index armhf packages, because we need one.
apt update
apt remove libraspberrypi0
apt install libraspberrypi0:armhf
 
### symlink the 32 bit libraries where they are looked for:
ln -s /usr/var/lib/arm-linux-gnueabihf /opt/vc/lib
 
### and now your bind mount needs to be:
/opt/vc/lib:/opt/vc/lib
 
### although you should be able to forego the symlink nonsense and simply use this bind mount:
/usr/var/lib/arm-linux-gnueabihf:/opt/vc/lib
 
### Some things i did that may or may not be necessary:
 
sudo chown root:video /dev/vchiq
sudo chmod g+rw /dev/vchiq
 
### The compose snippet that now works:
 
  jellyfin:
    image: linuxserver/jellyfin:arm32v7-latest
    container_name: jellyfin
    environment:
      - UID=1000 # The UID to run emby as (default: 2)
      - GID=1000 # The GID to run emby as (default 2)
    volumes:
      - /usr/var/lib/arm-linux-gnueabihf:/opt/vc/lib
      - ./config/jellyfin/:/config # Configuration directory
      - ./blah/blah/:/mnt/share0 # Media directory
    ports:
      - 8096:8096 # HTTP port
      - 8920:8920 # HTTPS port
    devices:
      - /dev/vchiq:/dev/vchiq # MMAL/OMX on Raspberry Pi
    restart: unless-stopped
### Note:
if you really need the 64 bit libraspberrypi0 on your host for some reason, simply move arm-linux-gnueabihf to safe keeping, and do the process in reverse, removing libraspberrypi0, dropping armhf support, updating apt, and reinstalling libraspberrypi0:aarch64  just make sure your bind mount points to wherever arm-linux-gnueabihf wound up.

**Comment by kls0e** [comment]
many thanks for your efforts. Would you be able to elaborate on the steps running raspberry pi OS 64-bit (beta 8/2020)? 

**EDIT: just soft-bricked my RPI4 running raspberry pi os following these steps, be careful.**

**Reply by whiteowl3** [reply]
i don't run raspberry pi OS, sorry.  my best guess would be to move the 32 bit libraries that you want to somewhere less disruptive (maybe within users home), and make sure to put libraspberrypi0 back how you found it (reinstall aarch64 version and remove armhf from dpkg) before you reboot, but these are only guesses.
